mlflow/metrics/genai/model_utils.py

```diff
@@ -150,11 +150,18 @@ def _call_deployments_api(deployment_uri, payload, eval_parameters, wrap_payload
         completion_inputs = {**payload, **eval_parameters}
         response = client.predict(endpoint=deployment_uri, inputs=completion_inputs)
         return _parse_chat_response_format(response)
-
+    elif endpoint_type is None:
+        # If the endpoint type is not specified, we don't assume any format
+        # and directly send the payload to the endpoint. This is primary for Databricks
+        # Managed Agent Evaluation, where the endpoint type may not be specified but the
+        # eval harness ensures that the payload is formatted to the chat format, as well
+        # as parsing the response.
+        chat_inputs = {**payload, **eval_parameters}
+        return client.predict(endpoint=deployment_uri, inputs=chat_inputs)
```

Comment:
Will an exception coming from this with an invalid interface be clear enough to users or do we need to wrap this with a try catch to re-throw with decorated exception text to point users to what they may have done wrong? 
Is there any mechanism for this creating a confusing experience?

Reply:
I think it is a good idea to wrap exception here. The original exception simply shows the actual error message from the endpoint, which might be clear for what was wrong, but users don't know what actual payload is passed to the endpoint because of do multiple conversion.

Reply:
Added an exception message like this:

<img width="754" alt="Screenshot 2024-09-25 at 11 49 23" src="https://github.com/user-attachments/assets/8d6ac2c6-cf9a-4066-97f0-99bd2b2570ae">